pkg/hakeeper/checkers/syshealth/check.go

```diff
@@ -0,0 +1,227 @@
+// Copyright 2021 - 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syshealth
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/hakeeper"
+	"github.com/matrixorigin/matrixone/pkg/hakeeper/checkers/util"
+	"github.com/matrixorigin/matrixone/pkg/hakeeper/operator"
+	hapb "github.com/matrixorigin/matrixone/pkg/pb/hakeeper"
+	"github.com/matrixorigin/matrixone/pkg/pb/logservice"
+)
+
+const (
+	defaultLogShardSize = 3
+)
+
+// Check checks system healthy or not.
+// If system wasn't healthy, we would generate
+// operators in order to shutdown all stores.
+func Check(
+	cluster hapb.ClusterInfo,
+	dnState hapb.DNState,
+	logState hapb.LogState,
+	currTick uint64,
+) ([]*operator.Operator, bool) {
+	sysHealthy := true
+
+	// parse all log stores for expired stores mainly
+	logStores := parseLogStores(logState, currTick)
+	if len(logStores.expired) == 0 {
+		return nil, sysHealthy
```

Comment:
`sysHealthy` always returns `true` here.

Reply:
You should review code from line 38 to line 57 again.
In line 42, if there was no expired log store, the system was healthy, no problem.
In line 50, detailed in this PR, if there was a log shard with more than half replicas expired, then consider the entire system as unhealthy, we would stop all stores in the cluster.